troposphere/static/js/components/images/common/Bookmark.react.js

```diff
@@ -27,26 +28,17 @@ export default React.createClass({
     },
 
     render: function() {
-        var image = this.props.image,
-            isFavorited = stores.ImageBookmarkStore.findOne({
+        let image = this.props.image;
+        let isFavorited = stores.ImageBookmarkStore.findOne({
```

Comment:
Get the ImageBookmark collection first. Then check if it has what you're looking for. `isFavorited` may be falsy, because it's in the middle of fetching bookmarks. (You may argue that the data is here because ..., it's not clear from this context, which is why it should to be fixed)